File: backend/Payments/src/Payments.Application/Common/Models/TransactionModel.cs
using AutoMapper;
using Payments.Application.Common.Extensions;
using Payments.Application.Common.Mappings;
using Payments.Domain.Entities;

namespace Payments.Application.Transactions.Queries.GetTransactionsWithPagination;
public class TransactionModel
{
    public long Id { get; set; }

    public string Icon { get; set; } = "/assets/images/icons/figma.png";

    public string Company { get; set; }

    public string Date { get; set; }

    public decimal Amount { get; set; }

    public TransactionStatus Status { get; set; }
}

File: backend/PaymentsNest/src/modules/transactions/models/transaction.model.ts
import { CurrentUser } from 'src/modules/user/user.model';
import { Transaction } from '../entities/transaction.entity';
import { timeAgo } from 'src/shared/date.utils';

export class TransactionModel {
  id: number;
  icon = '/assets/images/icons/figma.png';
  company: string;
  date: string;
  amount: number;
  status: number;

  constructor(user: CurrentUser, transaction: Transaction) {
    this.id = transaction.id;
    const isBuyer = user.companyId === transaction.buyerId;
    this.company = isBuyer ? transaction.seller.name : transaction.buyer.name;
    this.amount = isBuyer ? -transaction.amount : transaction.amount;
    this.date = timeAgo(transaction.created_date);
  }
}
